iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -171,6 +171,24 @@ public override async Task<Message> ReceiveAsync(TimeSpan timeout, CancellationT
                 {
                     break;
                 }
+                catch (OperationCanceledException)
+                {
+                    bool isObjectDisposedExceptionDisabled = false;
+#if !NET451
+                    AppContext.TryGetSwitch(AppContextConstants.DisableObjectDisposedExceptionForReceiveAsync, out isObjectDisposedExceptionDisabled);
+#endif
+                    // If cancellationToken.IsCancellationRequested is false, then this OperationCanceledException comes from 
```

Comment:
This is the wrong place to make this change - the new APIs should never change behavior and always throw the ODE.
The change should only exist for all `ReceiveAsync` variants except `ReceiveAsync` with `CancellationToken`.